src/modules/params/greeting.rs

```diff
@@ -1,17 +1,18 @@
 use anyhow::Result;
 
 pub fn get(args_toggle_greeting: bool, config_greet: Option<bool>) -> Result<bool> {
-	if !args_toggle_greeting && config_greet.is_none() {
-		return Ok(true);
+	if let Some(config_greet_value) = config_greet {
+		let greet = match args_toggle_greeting {
+			true => !config_greet_value,
+			_ => config_greet_value,
+		};
+		Ok(greet)
+	} else {
+		Ok(!args_toggle_greeting)
 	}
+}
 
-	let greet = match args_toggle_greeting {
-		true => !config_greet.unwrap(),
-		_ => config_greet.unwrap(),
-	};
```

Comment:
The preferred approach would be to extend the `unwrap()` function to fall back on the configuration defaults. 

This would be more uniform with the way such edge cases are handled in the rest of the application. It will also be a bit leaner, since we use the guard clause to exist early, and cleaner to read, since we can maintain a relatively low level of nesting without wrapping the entire function contents in a conditional statement.

```rs
let greet = match args_toggle_greeting {
	true => !config_greet.unwrap_or_else(|| Config::default().greeting.unwrap()),
	_ => config_greet.unwrap_or_else(|| Config::default().greeting.unwrap()),
};
```

the Config module probably needs to be imported.

Reply:
done